### PR TITLE
Death

### PR DESCRIPTION
**Pull request by iteloo**
Server-side logic still missing. Please review but not close yet.

### CODE REVIEW

**Comment by iteloo** [comment]
Forced push because forgot to pull+request before submitting PR

File: elm/Update.elm [hunk 1]
```diff
@@ -443,8 +431,42 @@ handleAction action model =
...
+            [ toGameServer Api.Death ]
+
+           else
+            []
+          )
+
+
+updateHealthWithAntihunger : GameCtx msg -> Upd GameModel
+updateHealthWithAntihunger ctx model =
+    if model.antihunger == 0 then
```
Comment (colin353) [inline elm/Update.elm:464]: I think antihunger shouldn't reset after it hits zero - just stay at zero and keep taking damage every day.

File: elm/View.elm [hunk 2]
```diff
@@ -51,8 +51,13 @@ gameView : GameModel -> Html GameMsg
 gameView model =
     div [ class "game-view" ] <|
         List.concat
-            [ [ topBar model
-              , div [ class "active-state" ]
+            [ [ topBar model ]
+            , -- [note] rounding error?
+              if model.health == 0 then
```
Comment (colin353) [inline elm/View.elm:56]: So you're showing the death stage when health = 0, but the aliveness is set by the Alive bool below? So what's the authoritative way to tell whether you're dead or not

File: server/game.go [hunk 3]
```diff
@@ -130,13 +131,14 @@ func (g *Game) RecieveMessage(user User, message Message) {
 		delete(g.UserSites, user)
 	case SetNameMessage:
 		user.SetName(msg.Name)
-
+	case DeathMessage:
+		user.SetAlive(false)
```
Comment (colin353) [inline server/game.go:135]: Should you also set health = 0?

Comment (colin353) [reply server/game.go:135]: Nevermind that's OK

File: elm/View.elm (hunk 2 above)
Comment (colin353) [reply elm/View.elm:56]: Pls ignore

File: server/game.go [hunk 4]
```diff
@@ -130,13 +131,14 @@ func (g *Game) RecieveMessage(user User, message Message) {
...
 	case TradeMessage:
 		isntSelfTrade := g.stagedUser != user
 		withinTimeInterval := g.GetTime()-g.stagingTime < TradeTimeout
 		log.Println("Trade proposed")
 		if isntSelfTrade && g.stagedUser != nil && withinTimeInterval {
-		log.Println("Trade accepted")
+			log.Println("Trade accepted")
```
Comment (colin353) [inline server/game.go:141]: Why indent weirdly

**Review by colin353 — APPROVED** [review]

**Review by colin353 — APPROVED** [review]

**Pull request merged**